runner-macros/src/entry.rs

```diff
@@ -0,0 +1,148 @@
+use proc_macro::TokenStream;
+use proc_macro2::Span;
+use quote::{quote, quote_spanned, ToTokens};
+
+enum Flavor {
+    Sandbox,
+    Testnet,
+    Mainnet,
+}
+
+fn parse_knobs(
+    mut input: syn::ItemFn,
+    args: syn::AttributeArgs,
+    is_test: bool,
+) -> Result<TokenStream, syn::Error> {
+    if input.sig.asyncness.take().is_none() {
+        let msg = "the `async` keyword is missing from the function declaration";
+        return Err(syn::Error::new_spanned(input.sig.fn_token, msg));
+    }
+
+    let mut flavor = None;
+    for arg in args {
+        match arg {
+            syn::NestedMeta::Meta(syn::Meta::NameValue(namevalue)) => {
+                let ident = namevalue
+                    .path
+                    .get_ident()
+                    .ok_or_else(|| {
+                        syn::Error::new_spanned(&namevalue, "Must have specified ident")
+                    })?
+                    .to_string()
+                    .to_lowercase();
+
+                let msg = format!("Not expected to received a value for: {}", ident.as_str());
+                return Err(syn::Error::new_spanned(namevalue, msg));
+            }
+            syn::NestedMeta::Meta(syn::Meta::Path(path)) => {
+                let name = path
+                    .get_ident()
+                    .ok_or_else(|| syn::Error::new_spanned(&path, "Must have specified ident"))?
```

Comment:
Why not default to `sandbox` so that a user can just write `#[runner::test]`?

Reply:
Good idea. I was thinking something like that too, but didn't immediately add that. Let me see if I can't just put it in